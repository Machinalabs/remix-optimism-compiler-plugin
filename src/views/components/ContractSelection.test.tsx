import React from "react"
import { render } from "@testing-library/react"

import {
  CompilationContractsType,
  ContractSelection,
  getContractsList,
} from "./ContractSelection"
import SampleResult from "./contractResultSample.json"

describe("ContractSelection tests", () => {
  test("snapshot", () => {
    const { container } = render(
      <ContractSelection
        contracts={(SampleResult as any) as CompilationContractsType}
      />
    )

    expect(container).toMatchSnapshot()
  })

  test("getContractsList", () => {
    const result = getContractsList(SampleResult as any)

    expect(result).toBeDefined()
    expect(result.length).toEqual(1)
    expect(result[0].file).toBeDefined()
    expect(result[0].name).toBeDefined()
  })

  test("component is rendered correctly", async () => {
    const { getByText } = render(
      <ContractSelection
        contracts={(SampleResult as any) as CompilationContractsType}
      />
    )

    const result = getContractsList(SampleResult as any)
    const selectName = `${result[0].name} (${result[0].file})`

    expect(getByText(`${selectName}`)).toBeInTheDocument()
  })
})
