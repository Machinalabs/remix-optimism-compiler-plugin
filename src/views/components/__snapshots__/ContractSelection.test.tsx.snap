// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContractSelection tests snapshot 1`] = `
<div>
  <select
    class="custom-select"
    data-id="compiledContracts"
    id="compiledContracts"
  >
    $
    <option
      value="Timer"
    >
      Timer (Timer.sol)
    </option>
  </select>
</div>
`;
